src/main/java/com/olgaboiar/mint/Router.java

```diff
@@ -3,43 +3,57 @@
 import com.olgaboiar.mint.handlers.*;
 
 import java.io.IOException;
-import java.util.HashMap;
-import java.util.Map;
-import java.util.Set;
+import java.util.*;
 
 public class Router {
     private Map<String, Map<String, IHandler>> routes = new HashMap<>();
+//    Map<String, IHandler> methodHandlers;
 
     public Router() {
         registerTestRoutes();
     }
 
     public Response route(Request request) throws IOException {
-        Map<String, IHandler> methodList = routes.get(request.getRequestedFile());
-        if (methodList == null) {
+        Map<String, IHandler> methodHandlers = routes.get(request.getUri());
+        if (methodHandlers == null) {
             return new NotFoundHandler().handleRequest(request);
         }
-        Set<String> allowedMethods = methodList.keySet();
-        IHandler handler =  methodList.getOrDefault(request.getMethod(), new NotAllowedHandler(allowedMethods));
+        String[] allowedMethods = new String[methodHandlers.keySet().size()];
```

Comment:
Consider using ArrayList instead of Array here.

Reply:
refactored in https://github.com/olgaboiar/MintHTTPServer/pull/9/commits/7109f1bf5b9a84b8c73ff201c49159ad8e04ceb8